core/example/benchmark/Cabana_CommPerformance.cpp

```diff
@@ -0,0 +1,586 @@
+        {
+            for ( int p = 0; p < send_per_neighbor; ++p )
+            {
+                export_ranks_host( num_stay + n * send_per_neighbor + p ) =
+                    neighbor_ranks[n+1];
+            }
+        }
+        auto export_ranks = Kokkos::create_mirror_view_and_copy(
+            data_memory_space(), export_ranks_host );
+
+        // Run tests and time the ensemble.
+        for ( int t = 0; t < num_run; ++t )
+        {
+            // Create source particles.
+            aosoa_type src_particles( num_particle );
+
+            // Create destination particles.
+            aosoa_type dst_particles;
+
+            // Create a distributor using the fast construction method.
+            distributor_fast_create.start(fraction);
+            auto comm_export_ranks =
+                Kokkos::create_mirror_view_and_copy(
+                    comm_memory_space(), export_ranks );
+            Cabana::Distributor<comm_memory_space> distributor_fast(
+                comm, comm_export_ranks, unique_ranks );
+            distributor_fast_create.stop(fraction);
+
+            // Create a distributor using the general construction method.
+            distributor_general_create.start(fraction);
+            comm_export_ranks =
+                Kokkos::create_mirror_view_and_copy(
+                    comm_memory_space(), export_ranks );
+            Cabana::Distributor<comm_memory_space> distributor_general(
+                comm, comm_export_ranks );
+            distributor_general_create.stop(fraction);
+
+            // Resize the destination aosoa.
+            dst_particles.resize( distributor_fast.totalNumImport() );
+
+            // Migrate the aosoa as a whole. Do host/device
+            // copies as needed.
+            distributor_aosoa_migrate.start(fraction);
+            auto comm_src_particles =
+                Cabana::Experimental::create_mirror_view_and_copy(
+                    comm_memory_space(), src_particles );
+            auto comm_dst_particles =
+                Cabana::Experimental::create_mirror_view(
+                    comm_memory_space(), dst_particles );
+            Cabana::migrate(
+                distributor_fast, comm_src_particles, comm_dst_particles );
+            Cabana::deep_copy( dst_particles, comm_dst_particles );
+            distributor_aosoa_migrate.stop(fraction);
+
+            // Migrate the aosoa using individual slices. Do host/device
+            // copies as needed.
+            distributor_slice_migrate.start(fraction);
+
+            comm_src_particles =
+                Cabana::Experimental::create_mirror_view_and_copy(
+                    comm_memory_space(), src_particles );
+            comm_dst_particles =
+                Cabana::Experimental::create_mirror_view(
+                    comm_memory_space(), dst_particles );
+
+            auto s0 = comm_src_particles.template slice<0>();
+            auto d0 = comm_dst_particles.template slice<0>();
```

Comment:
Probably should be using the free function here.

Reply:
Yes - I will use that now that it is in `master`